File: 기능개발.py
def solution(progresses, speeds):
    answer = []
    while progresses:
        cnt = 1
        progress = progresses.pop(0)
        speed = speeds.pop(0)
        
        if (100-progress) % speed:
            day = (100-progress) // speed + 1
        else:
            day = (100-progress) // speed
            
        for idx, val in enumerate(progresses):
            progresses[idx] = val + speeds[idx] * day
        
        if progresses == []:
            answer.append(cnt)
            break
        else:
            while progresses:
                if progresses[0] >= 100:
                    progresses.pop(0)
                    speeds.pop(0)
                    cnt += 1
                else:
                    break
        answer.append(cnt)
    return answer

progresses = [93,30,55]
speeds = [1,30,5]
print(solution(progresses, speeds))

File: 기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    const N = progresses.length
    let left = [];
    for (let i = 0; i < N; i++) {
        const progress = progresses[i];
        const speed = speeds[i];
        const time = Math.ceil((100 - progress) / speed);
        if (left.length === 0 || left[0] >= time) {
            left.push(time)
        } else {
            answer.push(left.length);
            left = [time];
        }
    }
    if (left.length > 0) {
        answer.push(left.length);
    }

    return answer;
}


console.log(solution([93,30,55], [1,30,5]))